File: ProtoType Pattern/PHP/Book.php
<?php


abstract class Book
{
    private $name;
    private $description;
    private $price;

    private $shippingFees;

    /**
     * Book constructor.
     * @param $name
     * @param $description
     * @param $price
     */
    public function __construct($name, $description, $price)
    {
        $this->name = $name;
        $this->description = $description;
        $this->price = $price;
    }

    /**
     * @return mixed
     */
    public function getShippingFees()
    {
        return $this->shippingFees;
    }

    /**
     * @param mixed $shippingFees
     */
    public function setShippingFees($shippingFees)
    {
        $this->shippingFees = $shippingFees;
    }


    public function __toString()
    {
        return "Book:{name: ".$this->name.",description: ".$this->description.",price: ".$this->price.",shippingFees: ".$this->shippingFees."}";
    }

}

File: ProtoType Pattern/Kotlin/src/main/kotlin/Book.kt
abstract class Book(var name:String,var description:String,var price:Int):Cloneable {

    private var shippingFees:Int = 0;

    fun setShippingFees(fees:Int){
        this.shippingFees = fees
    }


    override fun clone(): Book {
        return super.clone() as Book
    }

    override fun toString(): String {
        return "Book{name:${this.name},description:${this.description},price:${this.price},shippingFees:${shippingFees}}"
    }

}